Services/BackendService.cs

```diff
@@ -3,105 +3,63 @@ The backend service interacts
     with the backend
 */
 
+using System.Reflection.Emit;
 using Combat_Critters_2._0.Models;
 using CombatCrittersSharp;
-using CombatCrittersSharp.exception; //The wrapper class
-public static class BackendService
+using CombatCrittersSharp.exception;
+using Intents;
+namespace Combat_Critters_2._0.Services
 {
-    public static async Task<bool> LoginAsync(UserCredentials credentials)
+    public class BackendService
     {
+        private readonly IClient _client;
 
-        var client = new Client("http://api.combatcritters.ca:4000");
-        try
+        public BackendService(IClient client)
         {
-            Console.Write("Attempting to login...");
-            await client.Login(credentials.Username, credentials.Password);
-            Console.WriteLine("Login Success...");
-            return true;
+            _client = client; //Client is injected here
         }
-        catch(RestException e)
-        {
-            Console.WriteLine($"Failed to Login: {e.Message}");
-            return false;
-        }        
-    }
 
-    public static async Task<bool> CreateAccountAsync(Profile credentials)
-    {
-        // Interact with the Java backend to create a new account
-        //return await WrapperClient.CreateAccount(credentials);
-        return true;
-    }
+    
+        /// <summary>
+        /// This method sends a request to the backend for login
+        /// </summary>
+        /// <param name="credentials"></param>
+        /// <returns></returns>
+        public async Task<bool> LoginAsync(UserCredentials credentials)
```

Comment:
should this (and register) be returning bool, I feel like throwing an exception on failure makes more sense. that would probably need a global exception handler (display a toast if an exception gets that far) 

probably leave it for now and address this comment in a separate issue